src/include/sof/audio/component.h

```diff
@@ -230,10 +230,16 @@ enum {
 
 #endif /* #if defined(__ZEPHYR__) && defined(CONFIG_ZEPHYR_LOG) */
 
+#ifndef __ZEPHYR__
 #define comp_perf_info(pcd, comp_p)					\
-	comp_info(comp_p, "perf comp_copy peak plat %d cpu %d",		\
+	comp_info(comp_p, "perf comp_copy peak plat %u cpu %u",		\
 		  (uint32_t)((pcd)->plat_delta_peak),			\
 		  (uint32_t)((pcd)->cpu_delta_peak))
+#else
+#define comp_perf_info(pcd, comp_p)					\
+	comp_info(comp_p, "perf comp_copy peak plat %u",		\
+		(uint32_t)((pcd)->plat_delta_peak))
+#endif
```

Comment:
This is not good. We need to have same output on both OS'es.

There is a practical problem in that cavs_timer.c does not expose CCOUNT, so some extra work is needed to expose this in a portable manner. A temporary workaround would be to wrap a call to "__asm__ volatile ("rsr.CCOUNT %0" : "=r"(val));" in Zephyr builds, if platform is XTENSA. On XTOS this would call timer_get_system(cpu_timer_get().

Reply:
@serhiy-katsyuba-intel Checked with Zephyr folks and there is no standard interface to get CPU counter, so I suggest we add a SOF side interface to get the CPU cycle count and use that on both XTOS and Zephyr.

Reply:
@kv2019i , now both SOF timers, "system" and "platform", have been implemented to use Zephyr's API and both return wall clock (when compiled with Zephyr). What should we do: revert "system" timer API to return CCOUNT? So "platform" timer will be implemented using Zephyr API (wall clock) and "system" timer will be implemented in SOF using CCOUNT on XTENSA. That seems eliminate all ifdefs I've introduced in this PR.
Is my above understanding correct or you meant to add a third timer to SOF that will provide CCOUNT?

Reply:
I think we should add a separate interface to get CCOUNT. Zephyr provides a single clock interface and looking at SOF app code in src/audio, this seems sufficient for SOF application usage.  I.e. SOF application code can use the native Zephyr interfaces to get the job done (and for XTOS builds, we can map these on top of SOF clk.h like was done already).

So that leaves basicly the perf counters as a sole user that needs cycle count. I'd say it's easier we have a separate small interface for this.

Reply:
@serhiy-katsyuba-intel @kv2019i we should use native Zephyr APIs where we can to read the ticks/timers, but IIUC we are missing a Zephyr API to read CCOUNT today ? @andyross ?

Reply:
@lgirdwood exactly, I talked with Andy about this and there's no Zephyr API to read CCOUNT. so I think best path is to define interface on SOF side and use that.

Reply:
We should really define an Zephyr API for CCOUNT instead. SOF codebase should not be defining RTOS APIs.

Reply:
Well, to be clear, you can "`RSR(CCOUNT)`" anywhere in code you want right now, RSR() is an arch API that gets pulled in in essentially all the Zephyr headers (though I'm not a fan and it doesn't conform to our own naming rules, heh).   Blessing it further is problematic to use as a supported interface though; CCOUNT isn't a single clock, it varies from core to core and resets when they do (i.e. at suspend!).  It doesn't act like a "time" in the way that k_cycle_get_32() is required to.

Obviously it's still critically useful to Xtensa code doing microbenchmarks and tuning.  You just need to know what it is and how it works, which is a hard thing for an "OS" to specify.

If there's interest, we could absolutely rename this to e.g. "xtensa_ccount()" or whatever.  But coming up with a truly portable time API based on CCOUNT isn't likely to happen without some code being written to try to synchronize the cores and track suspend events, etc...

Reply:
@andyross we cant uses any xtensa specific APIs, it should be possible to have an RTOS API like ```k_cpu_get_cycles()```

Reply:
It clearly possible, it just requires that either we (1) document and accept the per-core behavior of the resulting API or (2) write code to provide a calibrated cross-core cycle output.  What we can't have (or rather, can't have portably) is an API that just returns CCOUNT, because that's not generically usable as a measure of "time".

Reply:
@andyross I'm good with either 1) or 2). As long as we have a portable Zephyr API where we can read the CPU cycles **count** (and it's understood these units are dynamic and changeable at runtime and not be be used for time keeping but can be used for counting).

Reply:
@lgirdwood @andyross It would be good to have an API to access a local cpu counter with no promise on the frequency stability or being globally synced, to be used by users only with understanding of the underlying hw implementation. Note it is not just local (per-core) but also may vary in time depending on the SoC specific integration of the cpu clock sources.